src/sio.ts

```diff
@@ -147,46 +159,14 @@ export class RPSIO {
         this.qspiGpioOutputEnable ^= value & GPIO_MASK;
         break;
       case DIV_UDIVIDEND:
-        this.divDividend = value >>> 0;
```

Comment:
Does it work correctly even though you use the same `this.divDividend = value;` for both the signed/unsigned cases (instead of `value >>> 0` vs `value | 0`)?

Reply:
in the end yes, it works in both cases.
I don't understand the reason to have separated registers, at the beginning I though putting a boolean flag to change the behavior between signed/unsigned, it turned out a useless handling.

Reply:
Have you tried dividing -3000 by 2?

In JavaScript I get different results (printing both results as signed, just to show the difference):

```js
const signedDiv = (a, b) => ((a|0)/(b|0))|0;
const unsignedDiv = (a, b) => ((a>>>0)/(b>>>0))|0;

console.log(signedDiv(-3000, 2)); // -1500
console.log(unsignedDiv(-3000, 2)); // 2147482148
```

Reply:
I added this test, it passes on both emu and sil.
what I've seen is that in the emu it doesn't matter if I use readUint32 or readInt32 it always return the right value.
test in the silicon instead it changes and it requires the right read method.
Seems that javascript (math.trunc?) automatically handles the right way in case of negative value.

Reply:
For the result, it shouldn't matter how you read it (you can convert between signed / unsigned at any stage).

But for the actual division, it seems to matter even if you use math.trunc:

```js
const signedDiv = (a, b) => Math.trunc((a|0)/(b|0))|0;
const unsignedDiv = (a, b) => Math.trunc((a>>>0)/(b>>>0))|0;

console.log(signedDiv(-3000, 2)); // -1500
console.log(unsignedDiv(-3000, 2)); // 2147482148
```

It could be the the current code is always doing signed division, and that's why the test passes. In this case, you can add a test for unsigned divide as well, trying to divide 0x80000000 by 2 using unsigned integer divide, and verify that you get the correct result (0x40000000):

```js
console.log((signedDiv(0x80000000, 2) >>> 0).toString(16)); // 0xc0000000
console.log((unsignedDiv(0x80000000, 2) >>> 0).toString(16)); // 0x40000000 
```

Reply:
arghhhh!!!  missing break forced signed code.
I've also had to force the readUint32 to return unsigned to pass this test:
`should perform a signed division 0x80000000 / 2 = 0xc0000000 REM 0`